### PR TITLE
IdStore

### PR DESCRIPTION
**Pull request by fmorency**
IdStore attribute interface, types, and mock unit tests.

### CODE REVIEW

**Review by hansl — APPROVED** [review]

**Comment by fmorency** [comment]
@hansl please review the new changes.

I used the workaround mentioned in https://github.com/rust-lang/cargo/issues/2911 so we can use `generate_random_eddsa_identity()` in 'many-framework`

File: src/many/src/server/module/_0_base.rs [hunk 1]
```diff
@@ -150,8 +150,7 @@ mod tests {
     use std::sync::{Arc, Mutex};
 
     use crate::{
-        protocol::Attribute, server::module::testutils::call_module,
-        types::identity::cose::tests::generate_random_eddsa_identity,
+        protocol::Attribute, server::module::testutils::call_module, types::identity::cose::testsutils::generate_random_eddsa_identity,
```
Comment (hansl) [inline src/many/src/server/module/_0_base.rs:153]: Why this change? I'm surprised this passes clippy.

Comment (fmorency) [reply src/many/src/server/module/_0_base.rs:153]: Needed to change from types::identity::cose::**tests**:: to types::identity::cose::**testsutils**::

It passes `fmt` and `clippy`

**Review by hansl — APPROVED** [review]
